Fix Mac compile errors

### Issue:
```
/Users/baidu/Documents/git_workspace/Paddle/paddle/utils/tests/test_CustomStackTrace.cpp:43:44: error: lambda capture 'countDown' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
                                          &countDown,
                                           ^
/Users/baidu/Documents/git_workspace/Paddle/paddle/utils/tests/test_CustomStackTrace.cpp:44:44: error: lambda capture 'layerSize' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
                                          &layerSize,
                                           ^
2 errors generated.
make[2]: *** [paddle/utils/tests/CMakeFiles/test_CustomStackTrace.dir/test_CustomStackTrace.cpp.o] Error 1
make[1]: *** [paddle/utils/tests/CMakeFiles/test_CustomStackTrace.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### Solution:

https://github.com/PaddlePaddle/Paddle/blob/71fa3ca9c4b1a7b5be569e32f8c2b5f659542f16/paddle/utils/tests/test_CustomStackTrace.cpp#L36-L49

Remove line 43 and line 44. For `countDown` and `layerSize` are `constexpr`,which means they can be used in lambda without being captured.

--- a/paddle/utils/tests/test_CustomStackTrace.cpp
+++ b/paddle/utils/tests/test_CustomStackTrace.cpp
@@ -12,10 +12,8 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */
 
-#include <chrono>
-
-#include <gflags/gflags.h>
-#include <gtest/gtest.h>
+#include <gflags/gflags.h>  // NOLINT
+#include <gtest/gtest.h>    // NOLINT
 
 #include "paddle/utils/CustomStackTrace.h"
 #include "paddle/utils/Locks.h"
@@ -39,14 +37,10 @@ void testNormalImpl(
   threads.reserve(FLAGS_test_thread_num);
 
   for (int32_t i = 0; i < FLAGS_test_thread_num; ++i) {
-    threads.emplace_back(new std::thread([&tracer,
-                                          &countDown,
-                                          &layerSize,
-                                          &startBarrier,
-                                          &doneBarrier,
-                                          &callback] {
-      callback(tracer, countDown, layerSize, startBarrier, doneBarrier);
-    }));
+    threads.emplace_back(
+        new std::thread([&tracer, &startBarrier, &doneBarrier, &callback] {
+          callback(tracer, countDown, layerSize, startBarrier, doneBarrier);
+        }));
   }
   size_t cntDown = countDown;
   while (cntDown-- > 0) {

